src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private StringBuilder builder;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom = stringToDate(dateFrom);
+        LocalDate localDateTo = stringToDate(dateTo);
+        LocalDate localDateFromRecord;
+        builder = new StringBuilder();
+        int salary;
+        builder.append("Report for period ");
+        builder.append(dateFrom).append(" - ");
+        builder.append(dateTo);
+        for (String name : names) {
+            salary = 0;
+            builder.append("\n").append(name).append(" - ");
+            for (String record : data) {
+                String[] recordArray = record.split(" ");
+                localDateFromRecord = stringToDate(recordArray[0]);
+                if (name.equalsIgnoreCase(recordArray[1])) {
+                    if ((localDateFromRecord.isAfter(localDateFrom)
+                            || localDateFromRecord.isEqual(localDateFrom))
+                            && (localDateFromRecord.isBefore(localDateTo)
+                            || localDateFromRecord.isEqual(localDateTo))) {
+                        salary += Integer.parseInt(recordArray[2])
+                                * Integer.parseInt(recordArray[3]);
```

Comment:
[Any magic numbers should be constants](https://github.com/mate-academy/jv-salary-info/blob/master/checklist.md#any-magic-numbers-should-be-constants)

Indexes are magic numbers
```suggestion
                localDateFromRecord = stringToDate(recordArray[0]);
                if (name.equalsIgnoreCase(recordArray[1])) {
                    if ((localDateFromRecord.isAfter(localDateFrom)
                            || localDateFromRecord.isEqual(localDateFrom))
                            && (localDateFromRecord.isBefore(localDateTo)
                            || localDateFromRecord.isEqual(localDateTo))) {
                        salary += Integer.parseInt(recordArray[2])
                                * Integer.parseInt(recordArray[3]);
```